go/cmd/dolt/commands/sqlserver/server.go

```diff
@@ -99,14 +105,37 @@ func Serve(ctx context.Context, version string, serverConfig ServerConfig, serve
 	dbNamesAndPaths := serverConfig.DatabaseNamesAndPaths()
 
 	if len(dbNamesAndPaths) == 0 {
-		var err error
-		mrEnv, err = env.DoltEnvAsMultiEnv(ctx, dEnv)
-		if err != nil {
-			return err, nil
+		if len(serverConfig.DataDir()) > 0 && serverConfig.DataDir() != "." {
```

Comment:
i'm not sure i understand why the logic is different between the two outer cases. A noop `fs.WithWorkingDir` that just sets it to "." seems fine? If we call `MultiEnvForDirectory` when we could have called `DoltEnvAsMultiEnv` that also seems ok?

Reply:
DoltEnvAsMultiEnv includes the dolt DB in the current working directory, as well as any in subdirectories of it. It's a subtle difference, but specifying data_dir = "." should exclude the dolt DB in the current working directory.

DoltEnvAsMultiEnv is really only there to support quick start, it makes some compromises in the data model.